### DISCUSSION TITLE
Update-ServiceNowRecord  Custom fields

### DISCUSSION THREAD

**Discussion by mgray40** [Snow-Shell/servicenow-powershell#163] [Q&A] [answered]
Is there any support for updating custom fields? There is the example of creating a new incident with custom fields but copying that syntax I still get the errors: 

Update-ServiceNowRecord -Values @AssetData and Update-ServiceNowRecord @AssetData both give
Update-ServiceNowRecord : Missing an argument for parameter 'Values'. Specify a parameter of type 'System.Collections.Hashtable' and try again.

attempting to do it inline 

"Update-ServiceNowRecord : The input object cannot be bound to any parameters for the command either because the command does not take pipeline input or the input and its properties do not match any of the 
parameters that take pipeline input." when attempting to update a record.

**Comment by gdbarron** [comment, marked as answer]
The setting of $AssetData wasn't included in your code, but assuming it's a hashtable....

`Update-ServiceNowRecord -Values @AssetData`, @ should be a $

`Update-ServiceNowRecord @AssetData`, $AssetData should be of the format
```powershell
$AssetData = @{
  'SysId' = '123456789fffffsdfsdfsf'
  'Values' = @{
    'field-to-update' = 'new-value'
  }
}
```
